tests/test_db_runtime_security.py

```
@@ -37,64 +38,86 @@
-        """
-        Prepare test fixtures by creating a temporary SQLite database file and initializing a DatabaseManager.
+@pytest.fixture
+def db_manager():
+    """
+    Create a temporary SQLite database file, initialize a DatabaseManager,
+    and yield both for use in tests. Guarantees cleanup on teardown.
+
+    Yields:
+        tuple[DatabaseManager, str]: (manager, db_path)
+    """
+    temp_db = tempfile.NamedTemporaryFile(delete=False, suffix=".db")
```

Ruff (0.15.12)

[warning] Use a context manager for opening files

(SIM115)